test/test.md

```diff
@@ -0,0 +1,22 @@
+# test
+test
+
+
+# test
+second test
+
+# test?
+
+# a `code block` in the header
+
+
```

Comment:
Other important test cases:

``` md
# Other punctuation, such as d.o.t.s, commas 

# Emdash –– and dash --

# Ampersand &

# Backslashes// or slashes\\

# Complex code blocks like `/foo/bar/:bazz?buzz=foo`

# Bold formatting **like this**

# Italic formatting _like this_

# All !@# the $%^ colors &*( of ){} the |~ punctuation < "' rainbow += 

# Seriously all of them ... Alt + [q-|] œ∑´®†¥¨ˆøπ“‘«
```

Might make more sense to have a more structured file (like a CSV of two columns `output,input`)  e.g. 

``` tsv
#test,# test
# 2 starts with number,#2-starts-with-number
```

Then you could generate the real `.md` file from that CSV as part of `npm pretest` or `npm posttest` (since we need it to verify what Github actually does since this algorithm is not part of the [official GFM specification](https://github.github.com/gfm/))